dockstore-client/src/main/java/io/github/collaboratory/wdl/WDLClient.java

```diff
@@ -170,92 +174,123 @@ public long launch(String entry, boolean isLocalEntry, String yamlRun, String js
+                        Map<String, String> outputJson = gson.fromJson(bracketContents, HashMap.class);
+                        System.out.println("Provisioning your output files to their final destinations");
+                        final List<String> outputFiles = bridge.getOutputFiles(localPrimaryDescriptorFile);
+                        FileProvisioning fileProvisioning = new FileProvisioning(abstractEntryClient.getConfigFile());
+                        List<ImmutablePair<String, FileProvisioning.FileInfo>> outputList = new ArrayList<>();
+                        for (String outFile : outputFiles) {
+                            // find file path from output
+                            final File resultFile = new File(outputJson.get(outFile));
+                            FileProvisioning.FileInfo new1 = new FileProvisioning.FileInfo();
+                            new1.setUrl(wdlOutputTarget + "/" + outFile);
+                            new1.setLocalPath(resultFile.getAbsolutePath());
+                            if (inputJson.containsKey(outFile + ".metadata")) {
+                                byte[] metadatas = Base64.getDecoder().decode((String)inputJson.get(outFile + ".metadata"));
+                                new1.setMetadata(new String(metadatas, StandardCharsets.UTF_8));
+                            }
+                            System.out.println("Uploading: " + outFile + " from " + resultFile + " to : " + new1.getUrl());
+                            outputList.add(ImmutablePair.of(resultFile.getAbsolutePath(), new1));
                         }
-                        System.out.println("Uploading: " + outFile + " from " + resultFile + " to : " + new1.getUrl());
-                        outputList.add(ImmutablePair.of(resultFile.getAbsolutePath(), new1));
+                        fileProvisioning.uploadFiles(outputList);
+                    } else {
+                        System.out.println("Output files left in place");
                     }
-                    fileProvisioning.uploadFiles(outputList);
-                } else {
-                    System.out.println("Output files left in place");
+                } catch (Exception e) {
+                    notificationsClient.sendMessage(NotificationsClient.PROVISION_OUTPUT, false);
+                    throw e;
+                }
+            } else {
+                WorkflowExecutionServiceApi clientWorkflowExecutionServiceApi = abstractEntryClient.getWorkflowExecutionServiceApi(wesUrl);
+                try {
+                    String tags = "WorkflowExecutionService";
+                    // Convert the filename to an array of bytes using a standard encoding
+                    byte[] descriptorContent = localPrimaryDescriptorFile.getAbsolutePath().getBytes(StandardCharsets.UTF_8);
+                    byte[] jsonContent = jsonRun.getBytes(StandardCharsets.UTF_8);
+
+                    //String jsonStrContent = null;
```

Comment:
Avoid checking in commented code unless there's a comment as to what (very likely) situation it would be uncommented in.